src/Microsoft.AspNetCore.Server.HttpSys/MessagePump.cs

```diff
@@ -32,7 +33,7 @@ internal class MessagePump : IServer
 
         private readonly ServerAddressesFeature _serverAddresses;
 
-        public MessagePump(IOptions<HttpSysOptions> options, ILoggerFactory loggerFactory)
```

Comment:
this mimics how anyone could hook their own IAuthenticationHandler on httpContext today, its always there today